examples/04_glm_first_level/plot_hrf.py

```diff
@@ -1,65 +1,163 @@
+which can for instance be useful when dealing with non human primate data
+acquired using a contrast agent. In our case, we input a custom response
+function for MION, a common
+`agent <https://en.wikipedia.org/wiki/MRI_contrast_agent>`_ used to
+enhance contrast on MRI images of monkeys.
 
-This example requires matplotlib.
-
-The hrf is the filter that couples neural responses to the metabolic-related
-changes in the MRI signal.  hrf models are simply phenomenological.
+The :term:`HRF` is the filter which couples neural responses to the
+metabolic-related changes in the MRI signal. :term:`HRF` models
+are simply phenomenological.
 
-In current analysis frameworks, the choice of hrf model is essentially left to
-the user. Fortunately, using the spm or Glover model does not make a huge
-difference.  Adding derivatives should be considered whenever timing
-information has some degree of uncertainty. It is actually useful to detect
+In current analysis frameworks, the choice of :term:`HRF` model is essentially
+left to the user. Fortunately, using the :term:`SPM` or
+Glover model does not make a huge
+difference. Adding derivatives should be considered whenever timing
+information has some degree of uncertainty, and is actually useful to detect
 timing issues.
 
+This example requires matplotlib and scipy.
 """
+
 #########################################################################
-# Set up some parameters for model display
-# ----------------------------------------
+# Define stimulus parameters and response models
+# ----------------------------------------------
 #
 # To get an impulse response, we simulate a single event occurring at time t=0,
 # with duration 1s.
 
 import numpy as np
-frame_times = np.linspace(0, 30, 61)
-onset, amplitude, duration = 0., 1., 1.
+
+time_length = 30.0
+frame_times = np.linspace(0, time_length, 61)
+onset, amplitude, duration = 0.0, 1.0, 1.0
 exp_condition = np.array((onset, duration, amplitude)).reshape(3, 1)
 
 #########################################################################
-# We sample this on a fris for display
+# Make a time array of this condition for display:
 stim = np.zeros_like(frame_times)
 stim[(frame_times > onset) * (frame_times <= onset + duration)] = amplitude
 
 #########################################################################
-# Now we have to define the candidate hrf models.
-hrf_models = [None, 'glover + derivative', 'glover + derivative + dispersion']
+# Define custom response functions for MION. Custom response
+# functions should at least take tr and oversampling as arguments:
+from scipy.stats import gamma
+
+
+def mion_response_function(tr, oversampling=16, onset=0.0):
+    """Implementation of the MION response function model.
+    Parameters
+    ----------
+    tr: float
+        scan repeat time, in seconds
+    oversampling: int, optional
+        temporal oversampling factor
+    onset: float, optional
```

Comment:
Do we need onset to ba given ? It should be 0 most of the times ?

Reply:
IMHO, being able to give the onset makes it easier to compute the derivative.
And you are right, it is set to 0 by default.